core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/FileAsyncResponseTransformer.java

```diff
@@ -44,13 +49,46 @@
     private volatile AsynchronousFileChannel fileChannel;
     private volatile CompletableFuture<Void> cf;
     private volatile ResponseT response;
+    private final long position;
+    private final FileTransformerConfiguration configuration;
 
     public FileAsyncResponseTransformer(Path path) {
         this.path = path;
+        this.configuration = FileTransformerConfiguration.defaultCreateNew();
+        this.position = 0L;
+    }
+
+    public FileAsyncResponseTransformer(Path path, FileTransformerConfiguration fileConfiguration) {
+        this.path = path;
+        this.configuration = fileConfiguration;
+        this.position = determineFilePositionToWrite(path);
+    }
+
+    private long determineFilePositionToWrite(Path path) {
+        if (configuration.fileWriteOption() == CREATE_OR_APPEND_EXISTING) {
+            try {
+                return Files.size(path);
+            } catch (NoSuchFileException e) {
+                // Ignore
+            } catch (IOException exception) {
+                throw SdkClientException.create("Cannot determine the current file size " + path, exception);
+            }
+        }
+        return  0L;
```

Comment:
Can we add a backlog item to revisit this? It makes me nervous to rely on catching an exception on a possibly hot code path.

Reply:
We could call `Files#exists` but it uses the same try-catch though... I'm not sure how we can optimize it, but yeah, I can add a TODO